packages/multi-select-combo-box/src/vaadin-multi-select-combo-box-mixin.js

```diff
@@ -758,11 +758,22 @@ export const MultiSelectComboBoxMixin = (superClass) =>
     __updateTopGroup(selectedItemsOnTop, selectedItems, opened) {
       if (!selectedItemsOnTop) {
         this._topGroup = [];
-      } else if (!opened) {
+      } else if (!opened || this.__needToSyncTopGroup()) {
         this._topGroup = [...selectedItems];
       }
     }
 
+    /** @private */
+    __needToSyncTopGroup() {
+      return (
+        this._topGroup &&
+        this._topGroup.some((item) => {
+          const selectedItem = this.selectedItems[this._findIndex(item, this.selectedItems, this.itemIdPath)];
+          return selectedItem && item !== selectedItem;
```

Comment:
This check for `selectedItem` will only work for object items but not string items, as in that case `selectedItem` will be `undefined` - I wonder if we should make it consistent? It's technically not required though.

Reply:
Ok, I noticed that "This fix does not change the intended behaviour of retaining the visible item order when the selected items are updated while the dropdown is opened". This would change if we also support string items.

Let's scope the implementation to object items then.

Reply:
Added a check and a brief comment. Do you think it would be enough?